docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+Event Command Format: `event n/[NAME] l/[LOCATION] d/DD/MM/YY HHmm - HHmm c/[COMMENTS]`
+
+*Step 5* +
+If the user supplies an `assignment` command, the `editAssignment()` method will be invoked. This method extracts the
+`assignmentName`, `moduleName`, `dateTime` and `comments` string to return a new instance of the `Assignment` class. +
+
+If the user supplies an `event` command, the `editEvent()` method will be invoked. This method extracts the
+`eventName`, `location`, `startDateTime`, `endDateTime` and `comments` string to return a new instance of the `Event` class.
+
+*Step 6* +
+This newly instanced class (either `Assignment` or `Event`) will be passed into the method `editTask()` of the `TaskList` class.
+The `editTask()` method of the `TaskList` class uses Java's `ArrayList` `set()` method to replace the task.
+
+*Step 7* +
+Finally, a `CommandResult` class is returned with `EDIT_SUCCESS_MESSAGE` passed as the parameter to the constructor of
+that class.
+
+The following sequence diagram summarises what happens when the `EditCommand` class is executed.
+
+image::EditCommand_SequenceDiagram.png[]
+
+==== Design Considerations
+* Placing invocation of new `Assignment` and `Event` class in `EditCommand` class
+** Rationale: +
+The `execute()` method of `EditCommand` class has to use the `Ui` class parsed as one of the parameters to get input from
+user on new details of the task. The new input captured will be then passed to the `editAssignment()` or `editEvent()` method
+in the `EditCommand` class.
+
+** Alternatives Considered: +
+The `editAssignment()` and `editEvent()` methods can be placed in the `Parser` class and called in the `prepareEditCommand`
+method of that class.
+
+
+* Using Java's `ArrayList#set()` method
+** Rationale: +
+When a task is selected to be edited, it is logical for the index of the task to not change as the task is being edited.
+Therefore, the `set()` method of `ArrayList` is used to replace the edited task with the old task.
+
+** Alternatives Considered: +
+Use the available `add` and `delete` methods, the new task is added into the list and the old task is deleted. However,
+this is not chosen as it is not intuitive for the user's task index to shift after editing the task.
+
+
+=== View Calendar feature
+
+[[calendar]]
+.Sample output of Calendar Command
+image::calendar2.png[]
+
+==== Implementation
+The `CalendarCommand` class extends the `Command` class with methods to implement the necessary pre-processing to display an overview of tasks in the given date.
+The following sequence diagram outlines an example execution of `CalendarCommand` when it is called and the interaction it has with the relevant components.
+
+.Interaction of CalendarCommand and the various major components
+image::calendar-diagram.png[]
+
+In particular, the diagram below shows the explicit execution flow that `CalendarCommand` takes.
+
+.Explicit execution flow of CalendarCommand
+image::addMonthlyCalendar.png[]
```

Comment:
Good description on this feature by using a sub sequence diagram to show the features ! 👍